pkg/operator/vttablet/pod.go

```diff
@@ -269,6 +269,7 @@ func UpdatePod(obj *corev1.Pod, spec *Spec) {
 	// Record a hash of desired tolerations to force the Pod to be recreated if
 	// one disappears from the desired list.
 	desiredStateHash.AddTolerations("tolerations", spec.Tolerations)
+	desiredStateHash.AddTopologySpreadConstraints("topologySpreadConstraints", spec.TopologySpreadConstraints)
```

Comment:
Oh, I didn't understand this is how it worked.  Can we make a note in the auto-genned docs above that touching the tolerations or topologySpreadConstraints spec will restart the pod.